### PR TITLE
Search cat

### PR DESCRIPTION
**Pull request by saurav-maharana**
- Added Cat Search API
- Only remote fetching and showing

### CODE REVIEW

**Comment by shounak-mulay** [comment]
Also, the CI is failing. 
Please complete the setup of scripts from the scripts readme to fix formatting and lint errors.

File: lib/repository/cat/cat_repository_impl.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class CatRepositoryImpl implements CatRemoteService {
+  late final CatRemoteService catRemoteService;
+
+  late final DomainCatMapper domainCatMapper;
+
+  CatRepositoryImpl({
+    required this.catRemoteService,
+    required this.domainCatMapper,
+  });
```
[the remark this answers is not in the document]
Comment (saurav-maharana) [reply lib/repository/cat/cat_repository_impl.dart:13]: Fixed.

File: lib/repository/cat/domain_cat_mapper.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter_template/domain/entity/cat/cat.dart';
+import 'package:flutter_template/foundation/mapper/mapper.dart';
+import 'package:flutter_template/services/entity/cat/remote/remote_cat.dart';
+
+abstract class DomainCatMapper extends Mapper<RemoteCat, Cat> {
+  // RemoteCat mapRemoteCat(RemoteCat from);
+
+  // List<RemoteCat> mapRemoteCatList(List<RemoteCat> from) =>
+  //     from.map((e) => mapRemoteCat(e)).toList();
```
[the remark this answers is not in the document]
Comment (saurav-maharana) [reply lib/repository/cat/domain_cat_mapper.dart:9]: Removed.

File: lib/repository/cat/cat_repository_impl.dart [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
-import 'package:flutter_template/services/cat/remote/cat_remote_service.dart';
 import 'package:flutter_template/services/entity/cat/remote/remote_cat.dart';
 
-class CatRepositoryImpl implements CatRemoteService {
-  late final CatRemoteService catRemoteService;
-
-  late final DomainCatMapper domainCatMapper;
+class CatRepositoryImpl implements CatRepository {
+  final CatRepository catRepository;
+  final DomainCatMapper domainCatMapper;
```
Comment (shounak-mulay) [inline lib/repository/cat/cat_repository_impl.dart:7]: I think we need to inject the remote service here.
Injecting the same class will cause a cyclic dependency.

File: lib/repository/cat/cat_repository_impl.dart [hunk 4]
```diff
@@ -1,22 +1,20 @@
...
-  Future<List<RemoteCat>> searchCat({required String searchTerm}) async {
-    final results = await catRemoteService.searchCat(
-      searchTerm: searchTerm,
-    );
-    return domainCatMapper.map(results);
+  Future<List<RemoteCat>> searchCat(String searchTerm) async {
+    final result = await catRepository.searchCat(searchTerm);
+
+    return result;
   }
```
Comment (shounak-mulay) [inline lib/repository/cat/cat_repository_impl.dart:19]: Same here. This is basically a recursion with no exit.

**Review by shounak-mulay — CHANGES_REQUESTED** [review]

File: lib/repository/cat/cat_repository_impl.dart (hunk 3 above)
Comment (saurav-maharana) [reply lib/repository/cat/cat_repository_impl.dart:7]: Checking on this front

File: lib/repository/cat/cat_repository_impl.dart (hunk 4 above)
Comment (saurav-maharana) [reply lib/repository/cat/cat_repository_impl.dart:19]: There is an error that is coming up with this. I've kept this for later fix.

**Review by alialh600pp — APPROVED** [review]

**Review by alialh600pp — CHANGES_REQUESTED** [review]
- [ ]

**Review by shounak-mulay — APPROVED** [review]

**Review by alialh600pp — COMMENTED** [review]
``

**Pull request merged**